main.go

```diff
@@ -405,77 +414,69 @@ func deploy(deployableItems []deployment.DeployableItem, config Config) (Artifac
 		PublicInstallPageURLs: map[string]string{},
 		PermanentDownloadURLs: map[string]string{},
 	}
-	errorCollection := []error{}
-	for _, apk := range apks {
-		log.Printf("Deploying apk file: %s", apk)
+	var errorCollection []error
+	var wg sync.WaitGroup
 
-		artifactURLs, err := uploaders.DeployAPK(apk, androidArtifacts, config.BuildURL, config.APIToken, config.NotifyUserGroups, config.NotifyEmailList, config.IsPublicPageEnabled)
-		if err != nil {
-			errorCollection = handleDeploymentFailureError(err, errorCollection)
-			continue
-		}
+	concurrency := determineConcurrency(config)
+	jobs := make(chan bool, concurrency)
+	combinedItems := append(append(apks, aabs...), others...)
+	mapLock := &sync.RWMutex{}
```

Comment:
nit: I'll leave it to you, but maybe `sync.Map` is a better choice here? https://pkg.go.dev/sync#Map

Reply:
Thanks for the tip as I did not know about this and I was directly searching for locks so google did not reward me with this result. 😄 

In this case I am leaning towards keeping the lock instead. My reasoning is that if I use `sync.Map` then the rest of the code also needs to be updated and also the rest of the code does not have a concurrent behaviour. 
The problem is with this struct:
```
artifactURLCollection := ArtifactURLCollection{
	PublicInstallPageURLs: map[string]string{},
	PermanentDownloadURLs: map[string]string{},
}
```
which is consumed everywhere in the subsequent stages. So based on these I think this time I will keep the lock.

Reply:
Oh that makes sense, let's keep the lock then for now.